packages/code-du-travail-frontend/src/outils/IndemniteLicenciement/ccn/1486/indemnite.js

```diff
@@ -0,0 +1,163 @@
+const COEFF_ETAM_1 = 0.3;
+const MAX_MONTH_INDEMNITE_ETAM = 10;
+const COEFF_IC = 1 / 3;
+const MAX_MONTH_INDEMNITE_IC = 12;
+const COEFF_CE = 1 / 5;
+const MAX_MONTH_INDEMNITE_CE = 7;
+
+function getAncienneteConventionnelle({
+  brancheCategorie,
+  ancienneteLegale,
+  hasBrancheContrat,
+  brancheContrats,
+  brancheAncienneteCE,
+  brancheAncienneteEnqueteur
+}) {
+  let ancienneteConventionnelle = 0;
+  if (brancheCategorie !== "CE") {
+    // categorie is ETAM or IC
+    ancienneteConventionnelle += parseFloat(ancienneteLegale);
+    if (hasBrancheContrat) {
+      ancienneteConventionnelle +=
+        brancheContrats.reduce(
+          (sumContrats, contrat) => sumContrats + parseInt(contrat.duration),
+          0
+        ) / 12; // contrat.duration is in months
+    }
+  } else {
+    ancienneteConventionnelle +=
+      parseInt(brancheAncienneteCE) / 12 + parseInt(brancheAncienneteEnqueteur);
+  }
+  return ancienneteConventionnelle;
+}
+
+function getSalaireRefConventionnelle({
+  anciennete,
+  salaire,
+  hasTempsPartiel = false,
+  salairePeriods = [],
+  hasSameSalaire = false,
+  salaires = []
+}) {
+  let moyenneSalaires = 0;
+
+  if (hasTempsPartiel) {
+    return salairePeriods.reduce(
+      (salaire, period) =>
+        parseFloat(salaire) +
+        (parseInt(period.salary, 10) * parseInt(period.duration, 10)) /
+          12 /
+          parseFloat(anciennete),
+      0
+    );
+  } else {
+    moyenneSalaires = hasSameSalaire
+      ? parseFloat(salaire)
+      : sum(salaires.map(a => parseFloat(a.salary))) / salaires.length;
+
+    return moyenneSalaires;
+  }
+}
+
+export function getIndemniteConventionnelle({
+  // Generiques
+  anciennete: ancienneteLegale,
+  salaire,
+  hasSameSalaire,
+  salaires,
+  hasTempsPartiel,
+  salairePeriods,
+  // Specifiques branches
+  brancheCategorie,
+  // Ancienneté ETAM/CI
+  hasBrancheContrat,
+  brancheContrats,
+  // Ancienneté CE
+  brancheAncienneteCE,
+  brancheAncienneteEnqueteur,
+  // Ajustement Salaire
+  hasBrancheNewSalaire,
+  hasBrancheNewRegularSalaire,
+  brancheNewRegularSalaire,
+  brancheNewIrregularSalaire
+}) {
+  // First of all, compute the new anciennete
+  const ancienneteConventionnelle = getAncienneteConventionnelle({
+    brancheCategorie,
+    ancienneteLegale,
+    hasBrancheContrat,
+    brancheContrats,
+    brancheAncienneteCE,
+    brancheAncienneteEnqueteur
+  });
+
+  let salaireRefConventionnelle;
+  let formuleConventionnelle;
+
+  if (hasBrancheNewSalaire) {
+    salaireRefConventionnelle = getSalaireRefConventionnelle({
+      hasTempsPartiel: false,
+      salaire: brancheNewRegularSalaire,
+      hasSameSalaire: hasBrancheNewRegularSalaire,
+      salaires: brancheNewIrregularSalaire
+    });
+  } else {
+    salaireRefConventionnelle = getSalaireRefConventionnelle({
+      anciennete: ancienneteLegale,
+      salaire,
+      hasTempsPartiel,
+      salairePeriods,
+      hasSameSalaire,
+      salaires
+    });
+  }
+
```

Comment:
let inputConventionels = {
  "salaire de référence (Sref): salaireRefConventionnel,
   "ancienneté conventionnelle (Ac)":  ancienneteConventionnelle
}

Reply:
? Is this for the formula ?

Reply:
it's for the details of the formula, yep